openapi/api.yml

```diff
@@ -150,6 +150,26 @@ paths:
                   $ref: '#/components/schemas/account'
         404:
           description: "No associated accounts found."
+  /accounts/search:
+    post:
```

Comment:
Any particular reason why you opted to make this a POST request? Thought reading data fits better a GET request. In hindsight, it could have potentially be the `get_accounts` with some extra parameters.

Reply:
Yeah it's a good question - normally idempotent requests would be a GET, you're right.

I prefer POST for 'search'-style requests because of a few generic reasons:

* they allow transmission of more data (before you hit weird things around how long URL query parameters can be)
* they prevent search terms being logged, especially if you're searching for specific email addresses (here we aren't - but maybe in the future this endpoint could)
* HTTP spec treats this as a single 'search' endpoint/resource - not a load of individual resources depending on what params are passed.

Basically - it's my preferred option over GET until we get something like [HTTP QUERY](https://httpwg.org/http-extensions/draft-ietf-httpbis-safe-method-w-body.html)